demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php

```diff
@@ -2364,43 +2407,31 @@ public function importStatementsAction(
         }
 
         // recreate uploaded array
-        $uploads = \explode(',', (string) $requestPost['uploadedFiles']);
+        $uploads = explode(',', (string) $requestPost['uploadedFiles']);
+        $files = array_map([$fileService, 'getFileInfo'], $uploads);
+        $importer = $importerFactory->createXlsxStatementImporter($excelImporter);
 
-        foreach ($uploads as $uploadHash) {
-            $file = $fileService->getFileInfo($uploadHash);
+        return $this->importStatements($files, $procedureId, $importer);
+    }
+
+    /**
+     * @param FileInfo[] $files
+     */
+    protected function importStatements(array $files, string $procedureId, XlsxStatementImport $importer): Response
+    {
+        foreach ($files as $file) {
             $fileName = $file->getFileName();
             try {
                 $importer->importFromFile($file);
 
                 if ($importer->hasErrors()) {
-                    return $this->renderTemplate(
-                        '@DemosPlanCore/DemosPlanProcedure/administration_excel_import_errors.html.twig',
-                        [
-                            'procedure'  => $procedureId,
-                            'context'    => 'statements',
-                            'title'      => 'statements.import',
-                            'errors'     => $importer->getErrorsAsArray(),
-                        ]
-                    );
+                    return $this->createErrorResponse($procedureId, $importer->getErrorsAsArray());
                 }
 
-                // on success:
-                $numberOfCreatedStatements = is_countable($importer->getCreatedStatements()) ? count($importer->getCreatedStatements()) : 0;
-                $this->getMessageBag()->add(
-                    'confirm',
-                    'confirm.statements.imported.from.xlsx',
-                    ['count' => $numberOfCreatedStatements, 'fileName' => $fileName]
-                );
-                $route = 'dplan_procedure_statement_list';
-                // Change redirect target if data input user
-                if ($permissions->hasPermission('feature_statement_data_input_orga')) {
-                    $route = 'DemosPlan_statement_orga_list';
-                }
+                $createdStatements = $importer->getCreatedStatements();
+                $numberOfCreatedStatements = is_countable($createdStatements) ? count($createdStatements) : 0;
```

Comment:
getCreatedStatements() returns always an array. Is it even possible to be not countable? Is this check necessary?